SU2_CFD/src/output/filewriter/CSTLFileWriter.cpp

```diff
@@ -0,0 +1,328 @@
+/*!
+ * \file CSTLFileWriter.cpp
+ * \brief STL Writer output class
+ * \author T. Kattmann, T. Albring
+ * \version 7.0.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/output/filewriter/CSTLFileWriter.hpp"
+#include "../../../include/output/filewriter/CParallelDataSorter.hpp"
+#include <iomanip> /*--- Used for ofstream.precision(int). ---*/
+
+
+const string CSTLFileWriter::fileExt = ".stl";
+
+CSTLFileWriter::CSTLFileWriter(vector<string> fields,
+                               unsigned short nDim,
+                               string fileName,
+                               CParallelDataSorter *dataSorter) :
+                CFileWriter(std::move(fields),
+                            std::move(fileName),
+                            dataSorter,
+                            fileExt,
+                            nDim){}
+
+
+CSTLFileWriter::~CSTLFileWriter(){}
+
+
+void CSTLFileWriter::Write_Data(){
+
+  /*--- This Write_Data routine has 3 major parts:
+    1. Prerequisite info: The parallel data-sorter distributes nodes of the primal mesh onto the processes

```

Comment:
Looks like you've already divided this into 3 logical tasks.. when you do the clean up, can you please decompose this into 3 smaller methods if possible? For functions/methods, smaller is better, and they should do 1 thing.

Reply:
I now made two separate functions for the first 2 points: `ReprocessElementConnectivity ` and `GatherCoordData`. In the `Write_Data() `method now the two functions are called and then Write_Data still has the code in it which actually writes the data to file.

Reply:
👍